admin/pre-commit

```diff
@@ -14,17 +14,35 @@ then
 fi
 SFILES=${SFILES:-$STAGED_FILES_CMD}
 
-echo "Checking PHP Lint..."
-for FILE in $SFILES
-do
-	php -l -d display_errors=0 "$PROJECT/$FILE"
+echo "Starting CodeIgniter precommit..."
+
+if [ "$SFILES" != "" ]
+then
+	echo "Linting PHP code..."
+	for FILE in $SFILES
+	do
+		php -l -d display_errors=0 "$PROJECT/$FILE"
+		if [ $? != 0 ]
+		then
+			echo "Fix the error(s) before commit."
+			exit 1
+		fi
+		FILES="$FILES $FILE"
+	done
+fi
+
+if [ "$FILES" != "" ]
+then
+	echo "Running PHPStan..."
+	# Run on whole codebase
+	./vendor/bin/phpstan analyse
+
 	if [ $? != 0 ]
 	then
-		echo "Fix the error before commit."
+		echo "Fix the phpstan error(s) before commit."
```

Comment:
phpstan doesn't fix the file, probably change wording with "Analyze code"

Reply:
I don't get it. Yes, phpstan doesn't fix the file. But if there are violations, it will echo them out as errors. So it must be fixed or resolved.

Reply:
oh, you're correct :+1: